### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by sfc-gh-zblackwood**
Replacing https://github.com/streamlit/gsheets-connection/pull/2 to resolve some git fork weirdness :)

### CODE REVIEW

**Comment by sfc-gh-zblackwood** [comment]
@sfc-gh-tszerszen
@sfc-gh-jcarroll

File: tests/test_public_sheet.py
Comment (sfc-gh-zblackwood) [inline tests/test_public_sheet.py:36]: Note that table name doesn't matter in the slightest -- it can be any string and the behavior is the same

File: tests/test_public_sheet.py
Comment (sfc-gh-zblackwood) [inline tests/test_public_sheet.py:45]: Note that with public sheets it is not possible to reference a specific worksheet by name, or at least I didn't find a way to accomplish that easily

File: tests/test_public_sheet.py
Comment (sfc-gh-zblackwood) [inline tests/test_public_sheet.py:66]: I would like to also test private sheets, but didn't want to bother getting github secrets setup for that for v1 of the tests

File: test-requirements.txt
Comment (sfc-gh-tszerszen) [inline test-requirements.txt:1]: Should not we add `mypy` to `test-requirements.txt`, although I've run `pip install -r test-requirements.txt`, I was not able to run `mypy --ignore-missing-imports .`, in my opinion this could be confusing. @sfc-gh-zblackwood what do you think?

File: .github/workflows/lint_and_test.yaml
Comment (sfc-gh-tszerszen) [inline .github/workflows/lint_and_test.yaml:23]: I think we should add `ruff`, `pytest` and `mypy` to test-requirements.txt

File: test-requirements.txt
Comment (sfc-gh-tszerszen) [reply test-requirements.txt:1]: Same comment is valid for ruff

File: .github/workflows/lint_and_test.yaml
Comment (sfc-gh-tszerszen) [inline .github/workflows/lint_and_test.yaml:3]: What is an expected outcome of adding this `.github/workflows/lint_and_test.yaml`? Will this be run by default on every push? I'm not sure for this file, so I'll double check with @sfc-gh-kbregula

**Review by sfc-gh-tszerszen — APPROVED** [review]
Overall LGTM and thank you for taking care of this @sfc-gh-zblackwood, this is an awesome work, test are present now and we have much clearer documentation ❤️ 

Please consider my comments and if you can please also wait for @sfc-gh-jcarroll and @sfc-gh-kbregula reviews for additional pair of eyes before merging 🙂

File: .github/workflows/lint_and_test.yaml
Comment (sfc-gh-kbregula) [inline .github/workflows/lint_and_test.yaml:3]:
```suggestion
  push:
    branches: ["main"]
  pull_request:
    branches: ["main"]
```
I think we want this to be triggered also when new PRs are created.

**Review by sfc-gh-jcarroll — APPROVED** [review]
Looks great to me

File: .github/workflows/lint_and_test.yaml
Comment (sfc-gh-zblackwood) [reply .github/workflows/lint_and_test.yaml:23]: I was going to say that "those are linting requirements more than test requirements", but since we do both at once, that probably does make sense

File: test-requirements.txt
Comment (sfc-gh-zblackwood) [reply test-requirements.txt:1]: Yup

**Comment by sfc-gh-zblackwood** [comment]
Accepted the workflow change, and added linters to test-reqs -- should be good to go now!

**Pull request merged**
